quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,167 @@
+import { findLastDotIndex } from "../../runner/services/autocomplete/highlight-and-score";
+import { TableInfo } from "../sql-context-evaluator";
+import { trinoToJs } from "./trinoToJs" 
+
+interface ObjectChild {
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    const childName = parentName ? `${parentName}.${key}` : key;
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+
+  return children.map(child => {
+    child.name = child.name.replace(/\.dataType$/, '');
```

Comment:
this replacement seems to be in the wrong place. It's function side effect
the function is called `getObjectChildren` but also we replace string inside, it's super unpredictable and can lead to increased investigation time if someone would need to find where and why the replacement is happening

Reply:
fixed